### PR TITLE
feat: Update react-router to v6.

### PR DESCRIPTION
**Pull request by yanguoyu**

### CODE REVIEW

**Comment by Keith-CY** [comment]
CI failed

File: packages/neuron-ui/src/states/stateProvider/actionCreators/settings.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { addNotification, addPopup } from './app'
 
 import { AppActions, StateDispatch } from '../reducer'
 
-export const createNetwork = (params: Controller.CreateNetworkParams) => (dispatch: StateDispatch, history: any) => {
+export const createNetwork = (params: Controller.CreateNetworkParams) => (dispatch: StateDispatch, navigate: any) => {
```
Comment (Keith-CY) [inline packages/neuron-ui/src/states/stateProvider/actionCreators/settings.ts:7]: `navigate` could be typed

**Review by Keith-CY — APPROVED** [review]

File: packages/neuron-ui/src/components/HardwareSign/index.tsx [hunk 2]
```diff
@@ -43,13 +43,13 @@ export interface HardwareSignProps {
   offlineSignType?: OfflineSignType
   onDismiss: () => void
   signMessage?: (password: string) => Promise<any>
-  history?: ReturnType<typeof useHistory>
+  navigate?: ReturnType<typeof useNavigate>
```
[the remark this answers is not in the document]
Comment (yanguoyu) [reply packages/neuron-ui/src/components/HardwareSign/index.tsx:46]: yeah, inertial thinking cause it.

File: packages/neuron-ui/src/components/ImportKeystore/index.tsx [hunk 3]
```diff
@@ -44,10 +44,10 @@ const ImportKeystore = () => {
   const {
     settings: { wallets },
   } = useGlobalState()
-  const history = useHistory()
+  const navigate = useNavigate()
   const [fields, setFields] = useState(defaultFields)
   const [openingFile, setOpeningFile] = useState(false)
-  const goBack = useGoBack(history)
+  const goBack = useGoBack(navigate)
```
[the remark this answers is not in the document]
Comment (yanguoyu) [reply packages/neuron-ui/src/components/ImportKeystore/index.tsx]:
```javascript
export const useGoBack = () => {
  const navigate = useNavigate()
  return useCallback(() => {
    navigate(-1)
  }, [navigate])
```
use `useGoBack` to cache the `() => { navigate(-1) }`

[the remark this answers is not in the document]
Comment (Keith-CY) [reply packages/neuron-ui/src/components/ImportKeystore/index.tsx]: Got it

**Review by Cedar67 — APPROVED** [review]
Verified.

**Pull request merged**